source/test/TestApp/PlayFabQoSTest.cpp

```diff
@@ -24,11 +25,29 @@ namespace PlayFabUnit
 {
 #if (!UNITY_IOS && !UNITY_ANDROID) && (!defined(PLAYFAB_PLATFORM_IOS) && !defined(PLAYFAB_PLATFORM_ANDROID) && !defined(PLAYFAB_PLATFORM_SWITCH))
     /// QoS API
-    void PlayFabQoSTest::QosResultApi(TestContext& testContext)
+    void PlayFabQoSTest::QoSResultApi(TestContext& testContext)
     {
+        LoginWithCustomIDRequest request;
+        request.CustomId = PlayFabSettings::buildIdentifier;
+        request.CreateAccount = true;
+
+        std::atomic_bool isLoggedIn = false;
+
+        PlayFabClientAPI::LoginWithCustomID(request,
+            [&isLoggedIn](const LoginResult& loginResult, void*) {
+                PlayFabSettings::staticPlayer->clientSessionTicket = loginResult.SessionTicket;
+                isLoggedIn.exchange(true);
+            },
+            [&isLoggedIn](const PlayFabError&, void*) {
+                isLoggedIn.exchange(true);
+            },
+                &testContext);
+
+        while (!isLoggedIn) {}
+
         QoS::PlayFabQoSApi api;
 
-        QoS::QoSResult result = api.GetQoSResult(5, 200);
+        QoS::QoSResult result = api.GetQoSResult(5);
 
         if (result.errorCode == 0)
```

Comment:
And this confirms it, the test passes or fails here, which means all internals are blocking until completion rather than ever return thread and timing control back to the test framework.